sr_tools/scripts/calib_strain_gauges.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+
+import os
+import rospy
+import numpy
+import csv
+from diagnostic_msgs.msg import DiagnosticArray
+
+
+class Calib_gauges():
+    def __init__(self, initial_weight, incremental_weight, final_weight):
+
+        self.initial_weight = initial_weight
+        self.incremental_weight = incremental_weight
+        self.final_weight = final_weight
+        self.motor_id = 0
+        self.number_of_tests = (self.final_weight-self.initial_weight)/self.incremental_weight
+        self.strain_gauges_id = 0
+        self.measurement = []
+        self.clean_measurement = []
+        self.headline_0 = []
+        self.temp_0 = []
+        self.headline_1 = []
+        self.temp_1 = []
+
+    def data_callback(self, data):
+
+        for i,value in enumerate(data.status): # iterate over the DiagnosticStatus message
+
+            fields = {}
+
+            for key,s in enumerate(value.values): # iterate over values array and store them in a dictionary with index and relative value
+
+                fields[i,s.key] = s.value
+           
+            for index, motor_id in fields.items(): # iterate over dictionary elements to search the wanted motor_id and the relative strain_gauge value
+
+                if fields.get((index[0],'Motor ID')) == "%s" % self.motor_id:
+                    if self.strain_gauges_id == 0:
+                        self.measurement.append(fields.get((index[0], 'Strain Gauge Left'))) #store read value in measurement array
+                    else:
+                        self.measurement.append(fields.get((index[0], 'Strain Gauge Right'))) #store read value in measurement array
+
+
+    def run_test(self):
+
+        self.motor_id = raw_input("insert motor id: ")
+
+        for self.strain_gauges_id in range(0, 2):
+
+           self.initial_weight = 0
+
+           for weight in range(0, self.number_of_tests):
+
+               self.initial_weight += self.incremental_weight
+
+               print("Apply %s g to strain_gauges %s" % (self.initial_weight, self.strain_gauges_id))
+               raw_input("Press enter when you are done...")
+
+               self.sub = rospy.Subscriber("/diagnostics_agg", DiagnosticArray, self.data_callback) # Subscribe diagnostic topic to collect data
```

Comment:
Do not declare a subscriber every time.
The subscriber should be declared only once in the constructor.